### PR TITLE
Bug 1779933: set version to start for s390/ppc until we have actual samples

### PR DESCRIPTION
**Pull request by gabemontero**
/assign @bparees 
/assign @adambkaplan

### CODE REVIEW

File: pkg/stub/handler.go [hunk 1]
```diff
@@ -588,6 +588,10 @@ func (h *Handler) Handle(event util.Event) error {
 		// Every time we see a change to the Config object, update the ClusterOperator status
 		// based on the current conditions of the Config.
 		cfg = h.refetchCfgMinimizeConflicts(cfg)
+		//TODO remove this setting of version once we start getting samples for z or ppc
+		if len(cfg.Spec.Architectures) > 0 && cfg.Spec.Architectures[0] != v1.AMDArchitecture && cfg.Spec.Architectures[0] != v1.X86Architecture {
```
Comment (bparees) [inline pkg/stub/handler.go]: i'm a little confused as to why this is specifically checking array index 0?

1) do we support multiple architectures being specified?
2) if we do, what happens if x86 is listed first and Z is listed second?

Comment (gabemontero) [reply pkg/stub/handler.go]: > i'm a little confused as to why this is specifically checking array index 0?
> 
>     1. do we support multiple architectures being specified?

at the moment, now ... we just have a place holder given this is an array to support it in the future

> 
>     2. if we do, what happens if x86 is listed first and Z is listed second?

we hard code the entry at start up, and the first time through should set all the clusteroperator related bits

Comment (gabemontero) [reply pkg/stub/handler.go]: adding validation type logic that handle a cluster admin replacing our boostrapping with incorrect choices is doable, but more complicated 

where we compare what they set in the array with what the golang arch env vars report, etc.

File: pkg/stub/handler.go [hunk 2]
```diff
@@ -588,6 +588,10 @@ func (h *Handler) Handle(event util.Event) error {
...
+			cfg.Status.Version = h.version
```
Comment (adambkaplan) [inline pkg/stub/handler.go]: Shouldn't we _always_ set this, and vary whether or not it is reported based on the `Available` status?

File: pkg/stub/handler.go (hunk 1 above)
Comment (adambkaplan) [inline pkg/stub/handler.go]: nit - move this to a shared method `isNonX86Arch(cfg)`

File: pkg/stub/handler.go (hunk 2 above)
Comment (gabemontero) [reply pkg/stub/handler.go]: no we shouldn't ... all that was sorted out in 4.0/4.1 initial dev

only report the version once you have achieved it

File: pkg/stub/handler.go (hunk 1 above)
Comment (gabemontero) [reply pkg/stub/handler.go]: ok I can do that

Comment (gabemontero) [reply pkg/stub/handler.go]: shared method added

**Comment by gabemontero** [comment]
> Since we need to backport this change and don't have e2e CI for Z/power, is there an easy way for us to add unit tests to verify we're setting version correctly?

Yes I believe so ... will start on that

**Comment by gabemontero** [comment]
unit test added @adambkaplan

File: pkg/stub/handler_test.go [hunk 3]
```diff
@@ -83,9 +84,17 @@ func TestWithDist(t *testing.T) {
...
+		t.Errorf("arch not set on bootstrap")
+	}
+	testValue := runtime.GOARCH
+	if testValue == v1.AMDArchitecture {
+		testValue = v1.X86Architecture
+	}
+	if cfg.Spec.Architectures[0] != testValue {
+		t.Errorf("arch set to %s instead of %s", cfg.Spec.Architectures[0], runtime.GOARCH)
 	}
+
```
Comment (adambkaplan) [inline pkg/stub/handler_test.go]: It doesn't look like `validate` verifies that we've set version correctly. Needs to be updated.

Comment (gabemontero) [reply pkg/stub/handler_test.go]: the setting of the version happens after the `setup()` call ... it does not need to be in `validate`

Comment (gabemontero) [reply pkg/stub/handler_test.go]: we don't need it everwhere

Comment (gabemontero) [reply pkg/stub/handler_test.go]: nevermind ... I'll do it everywhere

**Comment by gabemontero** [comment]
> @gabemontero need to check that version is set if we're reporting `Available=true; Progressing=false, Degraded=false`

we don't test that in the unit tests ... we already do in the e2e's

https://github.com/openshift/cluster-samples-operator/blob/master/test/e2e/cluster_samples_operator_test.go#L219-L242

too big of a change to do in the unit tests

File: pkg/stub/handler_test.go (hunk 3 above)
Comment (gabemontero) [reply pkg/stub/handler_test.go]: unit test updated

File: pkg/stub/handler_test.go [hunk 4]
```diff
@@ -98,6 +107,7 @@ func TestWithArchDist(t *testing.T) {
 		conditions,
 		statuses, t)
 
+
```
Comment (wking) [inline pkg/stub/handler_test.go]: nit: why do we keep adding empty lines?

**Pull request merged**